File: Week10/LifeBoat/kh.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int right = people.size() - 1;
    int left = 0;

    sort(people.begin(), people.end(),greater<int>());

    while (left < right + 1) {
        if (people[left] + people[right] <= limit)
            right--;
        left++;
        answer++;
    }
  
    return answer;
}

File: Week10/LifeBoat/kh.py
def solution(people, limit):
    answer = 0
    right=len(people) - 1
    left = 0
    people.sort(reverse=True)
    while left<right+1:
        if people[left]+people[right]<=limit:
            right-=1
        left+=1
        answer+=1

    return answer